examples/native-external-references-mockshop/functions/mockShop.ts

```diff
@@ -0,0 +1,181 @@
+import { FunctionEventContext } from '@contentful/functions-types';
+import {
+  EventHandler,
+  MappingHandler,
+  Product,
+  ProductLookupData,
+  QueryHandler,
+  ResourcesLookupHandler,
+  ResourcesSearchHandler,
+  SearchResultData,
+} from './types';
+
+const getMockShopUrl = (context: FunctionEventContext<Record<string, any>>) => {
+  const { apiEndpoint } = context.appInstallationParameters;
+  let mockShopUrl = apiEndpoint;
+  if (!mockShopUrl) {
+    mockShopUrl = 'https://mock.shop/api';
+    console.warn(`No API url configured, falling back to '${mockShopUrl}'`);
+  }
+  return mockShopUrl;
+};
+
+function withUrn(node: Product) {
```

Comment:
I feel like it would be a bit more clear if we move these utility functions to a `utils.ts`.

We could do this for `withUrn`, `withBadge` and `getMockShopUrl` so it does not clutter the main handler code

Reply:
I was on the fence to put these functions in `utils.ts`